From ef56db6c448d071414a6e8edc15ec04cbedcad2a Mon Sep 17 00:00:00 2001
From: Lion Yang <lion@aosc.io>
Date: Fri, 8 Feb 2019 18:53:40 +0800
Subject: [PATCH 1/3] client/cs_platform: do not support 32-bit on x64

Hardcode this to bypass the ugly thing which
scans your whole /lib directory.
---
 client/cs_platforms.cpp | 47 +----------------------------------------
 1 file changed, 1 insertion(+), 46 deletions(-)

diff --git a/client/cs_platforms.cpp b/client/cs_platforms.cpp
index 04ffd672e8..9fb5c43aed 100644
--- a/client/cs_platforms.cpp
+++ b/client/cs_platforms.cpp
@@ -153,52 +153,7 @@ void CLIENT_STATE::detect_platforms() {
             // If we're a 32 bit binary, then we obviously can.
             support32=1;
 #else
-            // If we're a 64 bit binary, the check is a bit harder.
-            // We'll use the file command to check installation of
-            // 32 bit libraries or executables.
-            const char *file[]={"/usr/bin/file","/bin/file",0};
-            const char *libdir[]={"/lib","/lib32","/lib/32","/usr/lib","/usr/lib32","/usr/lib/32"};
-            const int nlibdirs=sizeof(libdir)/sizeof(char *);
-
-            // find 'file'
-            //
-            eno=0;
-            while (1) {
-                if (boinc_file_exists(file[eno])) break;
-                eno++;
-                if (file[eno] == 0) break;
-            }
-
-            // now try to find a 32-bit C library.
-            if (file[eno] != 0) {
-                int i;
-                for (i=0; i < nlibdirs; i++) {
-                    struct dirent *entry;
-                    DIR *a = opendir(libdir[i]);
-                    // if dir doesn't exist, do the next one
-                    if (a == 0) continue;
-                    // dir exists. read each entry until you find a 32bit lib
-                    while ((support32 == 0) && ((entry=readdir(a)) != 0)) {
-                        strlcpy(cmdline, file[eno], 256);
-                        strlcat(cmdline, " -L ", 256);
-                        strlcat(cmdline, libdir[i], 256);
-                        strlcat(cmdline, "/", 256);
-                        strlcat(cmdline, entry->d_name, 256);
-                        f = popen(cmdline, "r");
-                        if (f) {
-                            while (!std::feof(f)) {
-                                if (!fgets(cmdline,256,f)) break;
-                                // If the library is 32-bit ELF, then we're
-                                // golden.
-                                if (strstr(cmdline, "ELF") && strstr(cmdline, "32-bit")) support32=1;
-                            }
-                            pclose(f);
-                        }
-                    }
-                    closedir(a);
-                    if (support32) break;
-                }
-            }
+            support32=0;
 #endif
         }
     }
-- 
2.39.1

